linker/shared/src/main/scala/org/scalajs/linker/frontend/optimizer/OptimizerCore.scala

```diff
@@ -2493,7 +2466,7 @@ private[optimizer] abstract class OptimizerCore(config: CommonPhaseConfig) {
           case ClassOf(ClassRef(_)) =>
             contTree(Null())
           case receiver =>
-            defaultApply(ClassType(ClassClass))
+            default
```

Comment:
This case still uses `default` after reading `newReceiver`. It should use `optTReceiver.get` instead.